perceval/backends/opnfv/functest.py

```diff
@@ -212,8 +216,18 @@ def _fetch(self, resource, params):
         logger.debug("Functest client requests: %s params: %s",
                      resource, str(params))
 
-        r = requests.get(url, params=params)
-        r.raise_for_status()
+        retries = 0
+
+        while retries < self.MAX_RETRIES:
+            try:
+                r = requests.get(url, params=params)
+                r.raise_for_status()
+                break
+            except requests.exceptions.ConnectionError as e:
+                time.sleep(0.5 * retries)
+                retries += 1
+        else:
+            raise e
```

Comment:
Really nice way to raise the exception if the number of retries have been consumed!